cmake/LYTestWrappers.cmake

```
@@ -261,9 +261,19 @@ function(ly_add_test)
     # Check to see whether or not this test target has been stored in the global list for walking by the test impact analysis framework
     get_property(all_tests GLOBAL PROPERTY LY_ALL_TESTS)
     if(NOT "${test_target}" IN_LIST all_tests)
+        # Extract the test target name from the namespace::target_name composite
+        string(REPLACE "::" ";" test_components ${test_target})
+        list(LENGTH test_components num_test_components)
+        if(num_test_components GREATER 1)
+            list(GET test_components 1 test_name)
+        else()
+            set(test_name ${test_components})
+        endif()
```

You can simplify this, if you know there's only ever at most 2 components and you always want the last one:
```suggestion
        string(REPLACE "::" ";" test_components "${test_target}")
        list(GET test_components -1 test_name)
```

Or just use a regex:
```suggestion
        string(REGEX REPLACE ".*::" "" test_name "${test_target}")
```

---

Nice, the regex version is much more succinct. Fixed 👍